server/app/views/components/breadcrumb/BreadcrumbFactory.java

```diff
@@ -0,0 +1,71 @@
+package views.components.breadcrumb;
+
+import static j2html.TagCreator.a;
+import static j2html.TagCreator.li;
+import static j2html.TagCreator.span;
+
+import com.google.common.collect.ImmutableList;
+import j2html.tags.ContainerTag;
+import j2html.tags.specialized.ATag;
+import j2html.tags.specialized.LiTag;
+import j2html.tags.specialized.NavTag;
+import j2html.tags.specialized.OlTag;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import views.components.Icons;
+
+/**
+ * Factory class for creating the breadcrumb UI component. See <a
+ * href="https://designsystem.digital.gov/components/breadcrumb/">USWDS Breadcrumb
+ * Documentation</a>.
+ */
+public final class BreadcrumbFactory {
+
+  /**
+   * Builds a breadcrumb UI component based off the given list.
+   *
+   * @param breadcrumbItems the list of items to include in the breadcrumb. Must be non-empty. The
+   *     order of items in the list determines the breadcrumb order.
+   * @throws IllegalArgumentException if {@code breadcrumbItems} is empty.
+   */
+  public NavTag buildBreadcrumb(ImmutableList<BreadcrumbItem> breadcrumbItems) {
```

Comment:
**nitpick (non-blocking):** Maybe something like `buildBreadcrumbTrail()` since it involves more than one breadcrumb?